api/src/main/java/org/apache/iceberg/Accessor.java

```diff
@@ -25,4 +25,7 @@ public interface Accessor<T> extends Serializable {
   Object get(T container);
 
   Type type();
+
+  /** Returns true if the current field or any ancestor in the access path is optional. */
+  boolean hasOptionalFieldInPath();
```

Comment:
```suggestion
  default boolean hasOptionalFieldInPath() { return false; }
```

this will avoid breaking the API and you'll be able to revert the revapi changes